jupyterlab/index.js

```diff
@@ -11,37 +11,31 @@ require('material-design-icons/iconfont/material-icons.css');
 
 require('jupyterlab/lib/default-theme/index.css');
 
-var lab = new JupyterLab();
-
-lab.registerPlugins([
-  require('jupyterlab/lib/about/plugin').aboutExtension,
-  require('jupyterlab/lib/clipboard/plugin').clipboardProvider,
-  require('jupyterlab/lib/commandpalette/plugin').commandPaletteProvider,
-  require('jupyterlab/lib/console/plugin').consoleExtension,
-  require('jupyterlab/lib/console/codemirror/plugin').rendererProvider,
-  require('jupyterlab/lib/csvwidget/plugin').csvHandlerExtension,
-  require('jupyterlab/lib/docregistry/plugin').docRegistryProvider,
-  require('jupyterlab/lib/editorwidget/plugin').editorHandlerProvider,
-  require('jupyterlab/lib/faq/plugin').faqExtension,
-  require('jupyterlab/lib/filebrowser/plugin').fileBrowserProvider,
-  require('jupyterlab/lib/help/plugin').helpHandlerExtension,
-  require('jupyterlab/lib/imagewidget/plugin').imageHandlerExtension,
-  require('jupyterlab/lib/inspector/plugin').inspectorProvider,
-  require('jupyterlab/lib/landing/plugin').landingExtension,
-  require('jupyterlab/lib/leafletwidget/plugin').mapHandlerExtension,
-  require('jupyterlab/lib/main/plugin').mainExtension,
-  require('jupyterlab/lib/mainmenu/plugin').mainMenuProvider,
-  require('jupyterlab/lib/markdownwidget/plugin').markdownHandlerExtension,
-  require('jupyterlab/lib/notebook/plugin').notebookTrackerProvider,
-  require('jupyterlab/lib/notebook/codemirror/plugin').rendererProvider,
-  require('jupyterlab/lib/rendermime/plugin').renderMimeProvider,
-  require('jupyterlab/lib/running/plugin').runningSessionsExtension,
-  require('jupyterlab/lib/services/plugin').servicesProvider,
-  require('jupyterlab/lib/shortcuts/plugin').shortcutsExtension,
-  require('jupyterlab/lib/terminal/plugin').terminalExtension
-
-
-  // require('jupyter-js-widgets-labextension/lib/plugin').widgetManagerExtension,
-]);
-
-window.onload = function() { lab.start(); }
+
+/**
+ * Validate an entry point given by the user.
+ */
+function validateEntryPoint(entryPoint) {
+  var data = jupyter.require(entryPoint);
+  if (!Array.isArray(data)) {
```

Comment:
even better, use the `__esModule` attribute to determine if we have an es6 module. This is added in typescript, as well as https://github.com/esnext/es6-module-transpiler/issues/86 and Babel.

```
if (data.__esModule) {
    data = data.default
}
```